docs/source/customizations/coprocessor.mdx

```diff
@@ -880,3 +817,85 @@ Your router considers all of the following scenarios to be a **failed response**
 - Your coprocessor responds with a non-`2xx` HTTP code.
 - Your coprocessor's response body doesn't match the JSON structure of the corresponding [request body](#example-requests-by-stage).
 - Your coprocessor's response body sets different values for [control properties](#property-reference) that must not change, such as `stage` and `version`.
+
+
+## Handling deferred query responses
+
+The Apollo Router supports the incremental delivery of query response data via [the `@defer` directive](../executing-operations/defer-support/):
+
+```mermaid
+sequenceDiagram
+  Client->>Router: Sends a query that<br/>defers some fields
+  Note over Router: Resolves non-deferred<br/>fields
+  Router->>Client: Returns data for<br/>non-deferred fields
+  Note over Router: Resolves deferred<br/>fields
+  Router->>Client: Returns data for<br/> deferred fields
+```
+
+For a single query with deferred fields, your router sends multiple "chunks" of response data to the client. If you enable coprocessor requests for the `RouterResponse` stage, your router sends a separate coprocessor request for _each chunk_ it returns as part of a deferred query.
+
+**Note the following about handling deferred response chunks:**
+
+- The [`status_code`](#status_code) and [`headers`](#headers) fields are included only in the coprocessor request for any response's _first_ chunk. These values can't change after the first chunk is returned to the client, so they're subsequently omitted.
+
+- If your coprocessor modifes the response [`body`](#body) for a response chunk, it must provide the new value as a _string_, _not_ as an object. This is because response chunk bodies include multipart boundary information in addition to the actual serialized JSON response data. [See examples.](#examples-of-deferred-response-chunks)
+  - Many responses will not contain deferred streams and for these the body string can usually be fairly reliably transformed into a JSON object for easy manipulation within the coprocessor. Coprocessors should be carefully coded to allow for the presence of a body that is not a valid JSON object.
+
+- Because the data is a JSON string at both `RouterRequest` and `RouterResponse`, it's entirely possible for a coprocessor to rewrite the body from invalid JSON content into valid JSON content. This is one of the primary use cases for `RouterRequest` body processing.
```

Comment:
I'm not 100% sure what this is conveying? It doesn't _seem_ specific to deferred queries given that _all_ operation bodies are strings at `RouterRequest` and `RouterResponse`, right?

Reply:
I'm trying to convey that if your response is not a result of a query which contains `@defer`, then you will only get one call to your coprocessor and that although the body is still a JSON String (see my comment above) it will deserialize into a JSON Object. However, you shouldn't assume that you can deserialize a body this way because as soon as your coprocessor has to handle a response to an `@defer` query, then it will fail.

The last paragraph is another reason why we are changing the body payload from JSON Object to JSON String.